### PR TITLE
Use ConcurrentHashMap in SegmentDirectoryLoaderRegistry and add test

### PR DESCRIPTION
**Pull request by npawar**
To handle `setSegmentDirectoryLoader` properly

### CODE REVIEW

**Review by walterddr — APPROVED** [review]
looks good. can you add a test for the loader init method?

**Comment by npawar** [comment]
On second thought, this change is tightly coupling the init of registry with pinot-server startup. This is breaking 
1. some places such as segment build, which are using the SegmentDirectoryLoaderRegistry agnostic to server startup
2. a whole bunch of tests which are using the registry/building segments and not worrying about registry init. 
As such I'm thinking of skipping this change, and external SegmentDirectoryLoader registrations can happen via "setSegmentDirectoryLoader" method.
Thoughts? cc @klsince since you're touching these methods lately

**Comment by klsince** [comment]
yeah, I feel `setSegmentDirectoryLoader` should be enough for now.

File: pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/loader/SegmentDirectoryLoaderRegistry.java [hunk 1]
```diff
@@ -37,14 +41,28 @@
  */
 public class SegmentDirectoryLoaderRegistry {
   private static final Logger LOGGER = LoggerFactory.getLogger(SegmentDirectoryLoaderRegistry.class);
-
   public static final String DEFAULT_SEGMENT_DIRECTORY_LOADER_NAME = "default";
-  private static final Map<String, SegmentDirectoryLoader> SEGMENT_DIRECTORY_LOADER_MAP = new HashMap<>();
+  private static final Map<String, SegmentDirectoryLoader> SEGMENT_DIRECTORY_LOADER_MAP = new ConcurrentHashMap<>();
```
Comment (klsince) [inline pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/loader/SegmentDirectoryLoaderRegistry.java:45]: this change of using concurrent hash map may be kept

Comment (npawar) [reply pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/loader/SegmentDirectoryLoaderRegistry.java:41]: done

**Comment by npawar** [comment]
> looks good. can you add a test for the loader init method?

added

**Review by mayankshriv — APPROVED** [review]

**Review by klsince — APPROVED** [review]

**Pull request merged**